atlasdb-timelock-server/src/main/java/com/palantir/atlasdb/timelock/atomix/InvalidatingLeaderProxy.java

```diff
@@ -50,13 +52,16 @@ private InvalidatingLeaderProxy(
     public static <T> T create(
             LocalMember localMember,
             DistributedValue<String> leaderId,
+            Election election,
             Supplier<T> delegateSupplier,
             Class<T> interfaceClass) {
         InvalidatingLeaderProxy<T> proxy = new InvalidatingLeaderProxy<>(
                 localMember,
                 leaderId,
                 delegateSupplier);
 
+        election.onElection(term -> proxy.clearDelegateUnchecked());
```

Comment:
It would only be an improvement, not required, but is there a way to only clearDelegate if the leadership changes as a result of the election. Basically if node A is leader, an election happens, and node A is still leader (maybe this isn't possible?) then I shouldn't needlessly invalidate locks.